### DISCUSSION TITLE
Missing space when using custom p2 touchpoint action

### DISCUSSION THREAD

**Discussion by cristianvalentincojocaru** [eclipse-equinox/p2#225] [General]
Hello everybody,

We have created a custom p2 touchpoint action that basically appends a preference in one of the settings file of our product. Unfortunately, we have encountered an issue regarding the parameters that we give to said action, more specifically, with the spaces at the start of our string value. It appends the value but completely ignores the space as the beginning of the String. We need the empty space for good functioning of our action.

Here is the action call: 
`instructions.configure=com.test.product.custom.action.native.appendPreference(scopeType:configuration,bundleID:rcp.modular.core,prefName:storedRepositories,prefValue: repo Configuration Tool \! http${#58}//www.website.com/lgfiles/updates/Eclipse/Product,createMissing:true,addDuplicate:false);`

And the code used to append the value:
```
boolean addDuplicate = Boolean.parseBoolean(addDuplicateParam);
if (addDuplicate || !prevPrefValue.contains(prefValueParam)) {
	//Update prevPrefValue to contain the new parameter given
	prefValueParam = prevPrefValue.concat(prefValueParam);
	// Save the new value to preferences
	pref.put(prefNameParam, prefValueParam);
}
pref.flush();
pref.sync();
```
Right after "prefValue" we give a string value that contains a space at the beginning. We have tried to enclose the string in **quotes (""), use unicode, hex code and css code values** but with no success. The only option we found is to hardcode a space but we do not really want to do that.

**Is there any reason why the whitespaces in the parameters are not taken into consideration? And if so, is there any way to add the space?**

**Comment by merks** [comment]
Does the value still look correct in the content.xml, e.g., like this:
```
      <touchpointData size='1'>
        <instructions size='2'>
          <instruction key='configure'>
            setProgramProperty(propName:eclipse.product,propValue:org.eclipse.oomph.setup.installer.product);setProgramProperty(propName:osgi.splashPath,propValue:platform${#58}/base/plugins/org.eclipse.oomph.setup.installer);setProgramProperty(propName:eclipse.application,propValue:org.eclipse.oomph.setup.installer.application);
          </instruction>
          <instruction key='unconfigure'>
            setProgramProperty(propName:eclipse.product,propValue:);setProgramProperty(propName:osgi.splashPath,propValue:);setProgramProperty(propName:eclipse.application,propValue:);
          </instruction>
        </instructions>
      </touchpointData>
```
You've tried ${#20} for encoding?

I really think someone (like you) will need to debug the process of parsing what's in the content.xml up to passing it to your custom touchpoint code to figure out what's going wrong...

**Reply by cristianvalentincojocaru** [reply]
Yes, the value does look okay in the context.xml file. We have also tried using the suggested encoding but with not success.
Thank you for your suggestion. We will further investigate the parsing mechanism but for now we went with the option to hardcode the empty space.
